packages/core/src/lib/sign-in-experience/sign-in.ts

```diff
@@ -47,56 +47,31 @@ export const validateSignIn = (
     })
   );
 
-  switch (signUp.identifier) {
-    case SignUpIdentifier.Username: {
-      assertThat(
-        signIn.methods.some(({ identifier }) => identifier === SignInIdentifier.Username),
-        new RequestError({
-          code: 'sign_in_experiences.miss_sign_up_identifier_in_sign_in',
-        })
-      );
-
-      break;
-    }
-
-    case SignUpIdentifier.Email: {
-      assertThat(
-        signIn.methods.some(({ identifier }) => identifier === SignInIdentifier.Email),
-        new RequestError({
-          code: 'sign_in_experiences.miss_sign_up_identifier_in_sign_in',
-        })
-      );
-
-      break;
-    }
-
-    case SignUpIdentifier.Sms: {
-      assertThat(
-        signIn.methods.some(({ identifier }) => identifier === SignInIdentifier.Sms),
-        new RequestError({
-          code: 'sign_in_experiences.miss_sign_up_identifier_in_sign_in',
-        })
-      );
-
-      break;
-    }
-
-    case SignUpIdentifier.EmailOrSms: {
-      assertThat(
-        signIn.methods.some(({ identifier }) => identifier === SignInIdentifier.Email) &&
-          signIn.methods.some(({ identifier }) => identifier === SignInIdentifier.Sms),
-        new RequestError({
-          code: 'sign_in_experiences.miss_sign_up_identifier_in_sign_in',
-        })
-      );
+  if (signUp.identifiers.includes(SignInIdentifier.Username)) {
```

Comment:
for (identifiers in signUp.identifiers) {}

Reply:
Good call, updated.
Also, check the `identifers.includes` pattern in other codes, and refactor the codes in the `core/src/lib/sign-in-experience/sign-up.ts`.